### PR TITLE
feat(policy): ability to skip default policies

### PR DESCRIPTION
**Pull request by johnharris85**
### Checklist prior to review

Fixes https://github.com/kumahq/kuma/issues/3346

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: api/mesh/v1alpha1/mesh.proto [hunk 1]
```diff
@@ -75,6 +75,10 @@ message Mesh {
   // Constraints that applies to the mesh and its entities
   Constraints constraints = 7;
 
+  // List of policies to skip creating by default when the mesh is created.
+  // e.g. TrafficPermission, MeshRetry, etc.
```
Comment (jakubdyszkiewicz) [inline api/mesh/v1alpha1/mesh.proto:79]: Add info that `*` means skip all policies

File: pkg/defaults/mesh/mesh_test.go [hunk 2]
```diff
@@ -71,4 +71,48 @@ var _ = Describe("EnsureDefaultMeshResources", func() {
...
+
+		// then default TrafficPermission doesn't exist
+		err = resManager.Get(context.Background(), core_mesh.NewTrafficPermissionResource(), core_store.GetByKey("allow-all-default", model.DefaultMesh))
+		Expect(core_store.IsResourceNotFound(err)).To(BeTrue())
+
+		// then default Retry doesn't exist
+		err = resManager.Get(context.Background(), core_mesh.NewRetryResource(), core_store.GetByKey("retry-all-default", model.DefaultMesh))
+		Expect(core_store.IsResourceNotFound(err)).To(BeTrue())
+
+		// then default TrafficRoute doesn't exist
```
Comment (jakubdyszkiewicz) [inline pkg/defaults/mesh/mesh_test.go:110]: does exist

File: api/mesh/v1alpha1/mesh.proto [hunk 3]
```diff
@@ -75,6 +75,10 @@ message Mesh {
...
+  repeated string skipCreatingInitialDefaultPolicies = 8;
```
[the remark this answers is not in the document]
Comment (johnharris85) [reply api/mesh/v1alpha1/mesh.proto:80]: Wanted to reflect that we only do this _at creation_. But if that's fine just being documented and not in the name of the field then I can change :)

**Review by slonka — COMMENTED** [review]
@johnharris85 made the build step pass, hope you don't mind (if you do I won't push to your branches again). Review-wise I agree with Jakub.

File: api/mesh/v1alpha1/mesh.proto (hunk 3 above)
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply api/mesh/v1alpha1/mesh.proto:80]: then maybe `skipCreatingInitialPolicies` ?
Feel free to pick the name you think is best

File: api/mesh/v1alpha1/mesh.proto (hunk 1 above)
Comment (johnharris85) [reply api/mesh/v1alpha1/mesh.proto]: Resolved

File: pkg/defaults/mesh/mesh.go [hunk 4]
```diff
@@ -83,3 +96,12 @@ func ensureDefaultResource(ctx context.Context, resManager manager.ResourceManag
...
 }
+
+func contains(s []string, str string) bool {
+	for _, v := range s {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
```
[the remark this answers is not in the document]
Comment (johnharris85) [reply pkg/defaults/mesh/mesh.go]: Resolved

File: pkg/defaults/mesh/mesh_test.go (hunk 2 above)
Comment (johnharris85) [reply pkg/defaults/mesh/mesh_test.go]: Resolved

**Comment by johnharris85** [comment]
All comments addressed :)

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**